src/main/kotlin/gg/skytils/skytilsmod/features/impl/misc/TrophyFishingProgress.kt

```diff
@@ -0,0 +1,258 @@
+/*
+ * Skytils - Hypixel Skyblock Quality of Life Mod
+ * Copyright (C) 2020-2023 Skytils
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published
+ * by the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+package gg.skytils.skytilsmod.features.impl.misc
+
+import gg.essential.universal.UResolution
+import gg.skytils.skytilsmod.Skytils
+import gg.skytils.skytilsmod.core.structure.GuiElement
+import gg.skytils.skytilsmod.utils.ItemUtil.getItemLore
+import gg.skytils.skytilsmod.utils.SBInfo
+import gg.skytils.skytilsmod.utils.SkyblockIsland
+import gg.skytils.skytilsmod.utils.Utils
+import gg.skytils.skytilsmod.utils.graphics.ScreenRenderer
+import gg.skytils.skytilsmod.utils.graphics.SmartFontRenderer
+import gg.skytils.skytilsmod.utils.graphics.colors.CommonColors
+import gg.skytils.skytilsmod.utils.stripControlCodes
+import net.minecraft.client.gui.inventory.GuiChest
+import net.minecraft.init.Items
+import net.minecraft.inventory.ContainerChest
+import net.minecraftforge.client.event.ClientChatReceivedEvent
+import net.minecraftforge.client.event.GuiScreenEvent
+import net.minecraftforge.fml.common.eventhandler.EventPriority
+import net.minecraftforge.fml.common.eventhandler.SubscribeEvent
+
+/**
+    TrophyFishingProgress.kt
+
+    A Kotlin class nested within an object by Erymanthus | RayDeeUx.
+
+    Intended to display missing/undiscovered Trophy Fishes. Until Hylin
+    rises from its grave, this is the closest thing I could concoct to a HUD
+    element that displays such information. Unfortunately requires prior
+    access to the Trophy Fishing menu (because, again, Hylin is somehow in
+    public archive as of writing this comment block).
+
+    Inspired by FluxCapacitor2's ContainerSellValue,
+    TrophyFishingProgress displays a player's current progress
+    in Trophy Fishing. This is in a separate file instead of being in MiscFeatures.kt
+    as this HUD display relies on multiple Forge events
+    (and so would be very out-of-place if inside MiscFeatures.kt).
+
+*   @author Erymanthus | RayDeeUx
+*/
+
+object TrophyFishingProgress {
+
+    private val trophyFishingProgressDisplay = TrophyFishingProgressDisplay() //do not delete this otherwise the HUD WONT SHOW
```

Comment:
From what I can tell this val is unused, so you should probably use an [init block](https://github.com/Skytils/SkytilsMod/blob/70575a0aedfa1639133baeb5c78f3b386db7bad0/src/main/kotlin/gg/skytils/skytilsmod/features/impl/dungeons/BossHPDisplays.kt#L50)?